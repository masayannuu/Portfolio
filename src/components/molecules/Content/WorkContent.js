import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Image } from 'reactbulma'

import ParagraphTitle from '../../atoms/Paragraph/ParagraphTitle'
import Modal from '../Modal/Modal'
import ModalLink from '../../atoms/Link/ModalLink'
import ModalBody from '../Modal/ModalBody'

const WrapDiv = styled.div`
  @media screen and (max-width: 768px) {
    width: 100%;
  }
  @media screen and (min-width: 769px) and (max-width: 1280px) {
    width: 593px;
  }
  @media screen and (min-width: 1281px) {
    width: 100%;
  }
`

const StyledTitle = styled(ParagraphTitle)`
  margin-bottom: 0.1rem;
`

const StyledSubTitle = styled(ParagraphTitle)`
  margin-bottom: 0.5rem;
`

const StyledDiv = styled.div`
  margin-top: 1rem;
`

const StyledImage = styled(Image)`
  &:hover {
    cursor: pointer;
  }
  width: 100%;
`

class WorkContent extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      content: props.content,
      showModal: false
    }
    this.handleOpenModal = this.handleOpenModal.bind(this)
    this.handleCloseModal = this.handleCloseModal.bind(this)
  }

  handleOpenModal () {
    this.setState({ showModal: true })
  }

  handleCloseModal () {
    this.setState({ showModal: false })
  }

  render () {
    const isTag = typeof this.state.content.information_tags !== 'undefined'
    return (
      <WrapDiv className="columns">
        <div className="column">
          <StyledTitle className="is-size-5 has-text-right">{this.state.content.title}</StyledTitle>
          <StyledSubTitle className="is-size-6 has-text-right">
            {this.state.content.sub_title}
          </StyledSubTitle>
          <div className="tk-a-otf-gothic-bbb-pr6n">
            <Modal
              handleCloseModal={this.handleCloseModal}
              showModal={this.state.showModal}
              title={this.state.content.title}
            >
              <ModalBody
                imageUrl={this.state.content.image_url}
                description={this.state.content.description}
                teamData={this.state.content.team_data}
                informationTags={this.state.content.information_tags}
                isTag={isTag}
              />
            </Modal>
          </div>
        </div>
        <div className="column">
          <StyledImage src={this.state.content.image_url} onClick={this.handleOpenModal} />
          <StyledDiv>
            <ModalLink onClick={this.handleOpenModal} />
          </StyledDiv>
        </div>
      </WrapDiv>
    )
  }
}

WorkContent.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string,
    sub_title: PropTypes.string,
    info_tags: PropTypes.array,
    image_url: PropTypes.string,
    description: PropTypes.string,
    team_data: PropTypes.shape({
      number: PropTypes.number,
      roll: PropTypes.string,
      work: PropTypes.string
    }).isRequired
  }).isRequired
}

export default WorkContent
