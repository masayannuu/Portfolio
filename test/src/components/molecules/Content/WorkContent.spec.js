import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'

import WorkContent from '../../../../../src/components/molecules/Content/WorkContent'

Enzyme.configure({ adapter: new Adapter() })

const tags = ['Ruby', 'Js', 'java', 'C#']
const info_tags = [{ name: 'Language', tags }, { name: 'SoftWare', tags }]

const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'

const team_data = {
  number: 3,
  roll: '設計・開発主担当',
  work:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ'
}

const content = {
  title: 'File Transfer Service',
  sub_title: 'ファイル転送サービス',
  image_url: 'https://bulma.io/images/placeholders/1280x960.png',
  description,
  team_data,
  info_tags
}

describe('Content WorkContent Molecules', () => {
  it('should render WorkContent', () => {
    const wrapper = Enzyme.shallow(<WorkContent content={content} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
