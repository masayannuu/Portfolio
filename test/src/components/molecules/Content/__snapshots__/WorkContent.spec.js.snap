// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content WorkContent Molecules should render WorkContent 1`] = `
<WorkContent__WrapDiv
  className="columns"
>
  <div
    className="column"
  >
    <WorkContent__StyledTitle
      className="is-size-5 has-text-right"
    >
      File Transfer Service
    </WorkContent__StyledTitle>
    <WorkContent__StyledSubTitle
      className="is-size-6 has-text-right"
    >
      ファイル転送サービス
    </WorkContent__StyledSubTitle>
    <div
      className="tk-a-otf-gothic-bbb-pr6n"
    >
      <Modal
        handleCloseModal={[Function]}
        showModal={false}
        title="File Transfer Service"
      >
        <ModalBody
          description="この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字"
          imageUrl="https://bulma.io/images/placeholders/1280x960.png"
          isTag={false}
          teamData={
            Object {
              "number": 3,
              "roll": "設計・開発主担当",
              "work": "この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ",
            }
          }
        />
      </Modal>
    </div>
  </div>
  <div
    className="column"
  >
    <WorkContent__StyledImage
      onClick={[Function]}
      src="https://bulma.io/images/placeholders/1280x960.png"
    />
    <WorkContent__StyledDiv>
      <ModalLink
        onClick={[Function]}
      />
    </WorkContent__StyledDiv>
  </div>
</WorkContent__WrapDiv>
`;
